### ISSUE TITLE
[Enhancement] When automatic updates are enabled, do not prevent the PC from being shut down and instead have the PC automatically shut down or restart after updates are installed while still allowing the user to cancel the scheduled shutdown

### ISSUE THREAD

**Issue by NintendoManiac64** [linuxmint/mintupdate#765]
**Is your feature request related to a problem? Please describe.**
If automatic updates are enabled then, if one is done using their PC, they cannot select "shut down" or "restart" and walk away while updates finish installing.

**Describe the solution you'd like**
When updates are automatically being installed, still allow the user shut down or restart their PC but what it will instead do is wait for the updates to finish installing and *then* shut down or restart.

But during this time when it is waiting, the user should be able to abort the shut down or restart in case they end up needing to use their PC again (basically avoiding the issue that Windows has where you straight-up cannot use your PC if you selected shut down or restart and it ends up installing automatic updates)

**Describe alternatives you've considered**
I suppose one could still put their PC into standby, but that still only just pauses the update installation and doesn't fully cut power.  I suppose there's always hibernate, but I'm guessing there's a reason why it seems to be disabled by default.

**Additional context**
Basically I was done using a laptop and was ready to shut it down and walk away from it, but I couldn't because the automatic updates prevented me from even selecting shut down.

**Comment by NintendoManiac64** [comment]
I apologize but I have not been able to revisit testing within the allotted 30 days if the basis of this enhancement request is even still an issue on modern versions of Mint. I hope it's not rude of me to ask if I may have a bit more time?

**Comment by mikerofone** [comment]
I've just retested this on Linux Mint 22.3 - Cinnamon 64-bit since I was hit with a long running automatic update (compiling NVidia drivers).

While the update is running, the Shutdown button is visible:

<img width="570" height="162" alt="Image" src="https://github.com/user-attachments/assets/ba737770-8aae-4cf4-9975-b433fe918bf9" />

However, clicking Shutdown brings up a prompt to enter the super user password:

<img width="800" height="688" alt="Image" src="https://github.com/user-attachments/assets/900afe30-7f92-48fd-8aca-668bab3a4cba" />
(Apologies for the screen photo, I cannot take a screenshot while this prompt is present.)


I'm not sure what happens when I enter the password. Will it abort the update gracefully and shut down? Will it force-kill the update and leave the system possibly broken? I didn't dare trying.
Instead, I clicked Cancel, which logs me out and drops me to the login screen, with the updates continuing to run in the background. I can log back in, which I did, to take these screenshots.
Just leaving the password prompt open will auto-cancel it about 10-20 seconds, dropping the user to the login screen.

But that still means that:

* There is no useful indication that there are updates currently running.
* There is no way to indicate "Finish updating, then shutdown/reboot".
* Clicking "Shutdown" gives a confusing prompt with unclear outcome, and drops the user to the login screen.

**Comment by Xoriander** [comment]
> But that still means that:
> 
>     * There is no useful indication that there are updates currently running.
> 
>     * There is no way to indicate "Finish updating, then shutdown/reboot".
> 
>     * Clicking "Shutdown" gives a confusing prompt with unclear outcome, and drops the user to the login screen.

Yes this could be really confusing for inexperienced users coming from Windows, hopefully this could be changed soon...

There  is a useful indication that there are updates currently running (the gear wheels):

<img width="75" height="53" alt="Image" src="https://github.com/user-attachments/assets/dbff5c45-50b1-427d-9c06-594e46167cbf" />

**Comment by Xoriander** [comment]
Ok I have looked into the process described by mikerofone a little bit deeper and it seems to be the behavior is not intended and there could be a bug.

The reason why the `Authentication required` dialog is shown up is the use of `systemd-inhibit` as root to block system shutdown while the automatic update operation is in progress (see [system_automatic_upgrades.py#L40](https://github.com/linuxmint/mintupdate/blob/master/usr/lib/linuxmint/mintUpdate/system_automatic_upgrades.py#L40)). But the shutdown button should be disabled by PolicyKit overrides (a temporary `pkla` file is symlinked to `/etc/polkit-1/localauthority/90-mandatory.d/` to prevent users from overriding the shutdown block via the GUI see [system_automatic_upgrades#L24-L25](https://github.com/linuxmint/mintupdate/blob/master/usr/lib/linuxmint/mintUpdate/system_automatic_upgrades.py#L24-L25) and [99-mintupdate-temporary.pkla](https://github.com/linuxmint/mintupdate/blob/master/usr/share/linuxmint/mintupdate/automation/99-mintupdate-temporary.pkla)).

The reason why the shutdown button is visible is that the package `polkitd-plka` is not installed by default (since Ubuntu 23.10).

@mtwebster  can you give a feedback please?
